gst/nnstreamer/tensor_common.c

```diff
@@ -979,45 +979,50 @@ gst_tensor_get_type (const gchar * typestr)
     return _NNS_END;
   }
 
-  if (type_string[0] == 'u' || type_string[0] == 'U') {
-    /**
-     * Let's believe the developer and the following three letters are "int"
-     * (case insensitive)
-     */
-    if (len == 6) {             /* uint16, uint32 */
-      if (type_string[4] == '1' && type_string[5] == '6')
+  if (g_regex_match_simple ("^uint(8|16|32|64)$",
+          type_string, G_REGEX_CASELESS, 0)) {
+    size = g_ascii_strtoull (&type_string[4], NULL, 10);
+
+    switch (size) {
+      case 8:
```

Comment:
How about use gst_tensor_get_element_size() instead of a hard-coded value?

Reply:
Nope. ```gst_tensor_get_element_size()``` is a reverse function of what's needed here.
```gst_tensor_get_element_size()```: tensor_type --> size
what we need here: size --> tensor_type

Reply:
@myungjoo Ah! Got it. My bad.